boards/shields/m5stack_m5dial/m5stack_m5dial.overlay

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/dt-bindings/pwm/pwm.h>
+#include <zephyr/dt-bindings/display/panel.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	chosen {
+		zephyr,display = &gc9x01x_lcd;
+		zephyr,display-backlight = &backlight_pwm;
+		zephyr,rtc = &pfc8563_rtc;
+	};
+
+	aliases {
+		sw0 = &user_button_1; /* overwrite StampS3 user button */
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+
+		user_button_1: button_1 {
+			label = "User button 1";
+			gpios = <&m5stack_stamps3_header 24 GPIO_ACTIVE_LOW>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+
+		rotary_left: rotary_left {
+			label = "rotary left";
+			gpios = <&m5stack_stamps3_header 20 GPIO_ACTIVE_LOW>;
+			zephyr,code = <INPUT_KEY_LEFT>;
+		};
+
+		rotary_right: rotary_right {
+			label = "rotary right";
+			gpios = <&m5stack_stamps3_header 22 GPIO_ACTIVE_LOW>;
+			zephyr,code = <INPUT_KEY_RIGHT>;
+		};
```

Comment:
rotary encoder != regular gpio keys AFAIK, you should use gpio-qdec driver

```
	qdec {
		compatible = "gpio-qdec";
		gpios = <&m5stack_stamps3_header 20 GPIO_ACTIVE_LOW>,
		        <&m5stack_stamps3_header 22 GPIO_ACTIVE_LOW>;
		steps-per-period = <4>;
		zephyr,axis = <INPUT_REL_WHEEL>;
		sample-time-us = <2000>;
		idle-timeout-ms = <200>;
	};
```


and then works when used for a lvgl encoder:

```
	lvgl_encoder_input {
		compatible = "zephyr,lvgl-encoder-input";
		rotation-input-code = <INPUT_REL_WHEEL>;
		button-input-code = <INPUT_KEY_0>;
	};
```